sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -1272,17 +1272,22 @@ private void validateConfig() {
         URI uri;
         try {
             uri = new URI(serviceEndpoint);
+            if (!Strings.isNullOrEmpty(uri.getPath()) || !Strings.isNullOrEmpty(uri.getQuery())) {
+                serviceEndpoint = uri.getScheme() + "://" + uri.getAuthority();
+                uri = new URI(serviceEndpoint);
+            }
         } catch (URISyntaxException e) {
             throw new IllegalArgumentException("invalid serviceEndpoint", e);
         }
 
         if (preferredRegions != null) {
             // validate preferredRegions
+            URI finalUri = uri;
```

Comment:
Not sure why this is needed, but it doesn't do any harm either.

Reply:
Local variables used in lambda expressions need to be final or effectively final, so because the Uri is not effectively final anymore had to copy it over to new variable.